### ISSUE TITLE
CentOS 7 build is broken

### ISSUE THREAD

**Issue by tomzo** [gocd-contrib/gocd-oss-cookbooks#47]
Running docker image build for centos 7:
```
docker build . -t gocddev/gocd-dev-build:centos7 -f Dockerfile.centos7
```
Results in following errors:
```
================================================================================
 Package                     Arch   Version                Repository      Size
================================================================================
Installing:
 autoconf                    noarch 2.69-11.el7            base           701 k
 bison                       x86_64 3.0.4-2.el7            base           674 k
 bzip2-devel                 x86_64 1.0.6-13.el7           base           218 k
 cmake                       x86_64 2.8.12.2-2.el7         base           7.1 M
 devtoolset-6-gcc            x86_64 6.3.1-3.1.el7          centos-sclo-rh  28 M
 devtoolset-6-gcc-c++        x86_64 6.3.1-3.1.el7          centos-sclo-rh  11 M
 flex                        x86_64 2.5.37-6.el7           base           293 k
 glibc-devel                 x86_64 2.17-292.el7           base           1.1 M
 kernel-devel                x86_64 3.10.0-1062.1.1.el7    updates         18 M
 libcurl-devel               x86_64 7.29.0-54.el7          base           302 k
 libedit-devel               x86_64 3.0-12.20121213cvs.el7 base            32 k
 libffi-devel                x86_64 3.0.13-18.el7          base            23 k
 libxml2-devel               x86_64 2.9.1-6.el7_2.3        base           1.0 M
 libxslt-devel               x86_64 1.1.28-5.el7           base           309 k
 libyaml-devel               x86_64 0.1.4-11.el7_0         base            82 k
 make                        x86_64 1:3.82-24.el7          base           421 k
 openssl-devel               x86_64 1:1.0.2k-19.el7        base           1.5 M
 readline-devel              x86_64 6.2-11.el7             base           139 k
 zlib-devel                  x86_64 1.2.7-18.el7           base            50 k
Updating:
 bash                        x86_64 4.2.46-33.el7          base           1.0 M
Installing for dependencies:
 audit-libs-python           x86_64 2.8.5-4.el7            base            76 k
 checkpolicy                 x86_64 2.5-8.el7              base           295 k
 devtoolset-6-binutils       x86_64 2.27-12.el7.1          centos-sclo-rh 5.7 M
 devtoolset-6-libstdc++-devel
                             x86_64 6.3.1-3.1.el7          centos-sclo-rh 2.5 M
 devtoolset-6-runtime        x86_64 6.1-1.el7              centos-sclo-rh  20 k
 glibc-headers               x86_64 2.17-292.el7           base           687 k
 groff-base                  x86_64 1.22.2-8.el7           base           942 k
 kernel-headers              x86_64 3.10.0-1062.1.1.el7    updates        8.7 M
 keyutils-libs-devel         x86_64 1.5.8-3.el7            base            37 k
 krb5-devel                  x86_64 1.15.1-37.el7_7.2      updates        272 k
 libarchive                  x86_64 3.1.2-12.el7           base           319 k
 libcgroup                   x86_64 0.41-21.el7            base            66 k
 libcom_err-devel            x86_64 1.42.9-16.el7          base            32 k
 libedit                     x86_64 3.0-12.20121213cvs.el7 base            92 k
 libgcrypt-devel             x86_64 1.5.3-14.el7           base           129 k
 libgomp                     x86_64 4.8.5-39.el7           base           158 k
 libgpg-error-devel          x86_64 1.12-3.el7             base            16 k
 libkadm5                    x86_64 1.15.1-37.el7_7.2      updates        178 k
 libmpc                      x86_64 1.0.1-3.el7            base            51 k
 libselinux-devel            x86_64 2.5-14.1.el7           base           187 k
 libselinux-python           x86_64 2.5-14.1.el7           base           235 k
 libselinux-utils            x86_64 2.5-14.1.el7           base           151 k
 libsemanage-python          x86_64 2.5-14.el7             base           113 k
 libsepol-devel              x86_64 2.5-10.el7             base            77 k
 libverto-devel              x86_64 0.2.5-4.el7            base            12 k
 libxslt                     x86_64 1.1.28-5.el7           base           242 k
 libyaml                     x86_64 0.1.4-11.el7_0         base            55 k
 lzo                         x86_64 2.06-8.el7             base            59 k
 m4                          x86_64 1.4.16-10.el7          base           256 k
 mpfr                        x86_64 3.1.1-4.el7            base           203 k
 ncurses-devel               x86_64 5.9-14.20130511.el7_4  base           712 k
 pcre-devel                  x86_64 8.32-17.el7            base           480 k
 perl                        x86_64 4:5.16.3-294.el7_6     base           8.0 M
 perl-Carp                   noarch 1.26-244.el7           base            19 k
 perl-Data-Dumper            x86_64 2.145-3.el7            base            47 k
 perl-Encode                 x86_64 2.51-7.el7             base           1.5 M
 perl-Exporter               noarch 5.68-3.el7             base            28 k
 perl-File-Path              noarch 2.09-2.el7             base            26 k
 perl-File-Temp              noarch 0.23.01-3.el7          base            56 k
 perl-Filter                 x86_64 1.49-3.el7             base            76 k
 perl-Getopt-Long            noarch 2.40-3.el7             base            56 k
 perl-HTTP-Tiny              noarch 0.033-3.el7            base            38 k
 perl-PathTools              x86_64 3.40-5.el7             base            82 k
 perl-Pod-Escapes            noarch 1:1.04-294.el7_6       base            51 k
 perl-Pod-Perldoc            noarch 3.20-4.el7             base            87 k
 perl-Pod-Simple             noarch 1:3.28-4.el7           base           216 k
 perl-Pod-Usage              noarch 1.63-3.el7             base            27 k
 perl-Scalar-List-Utils      x86_64 1.27-248.el7           base            36 k
 perl-Socket                 x86_64 2.010-4.el7            base            49 k
 perl-Storable               x86_64 2.45-3.el7             base            77 k
 perl-Text-ParseWords        noarch 3.29-4.el7             base            14 k
 perl-Time-HiRes             x86_64 4:1.9725-3.el7         base            45 k
 perl-Time-Local             noarch 1.2300-2.el7           base            24 k
 perl-constant               noarch 1.27-2.el7             base            19 k
 perl-libs                   x86_64 4:5.16.3-294.el7_6     base           688 k
 perl-macros                 x86_64 4:5.16.3-294.el7_6     base            44 k
 perl-parent                 noarch 1:0.225-244.el7        base            12 k
 perl-podlators              noarch 2.5.1-3.el7            base           112 k
 perl-threads                x86_64 1.87-4.el7             base            49 k
 perl-threads-shared         x86_64 1.43-6.el7             base            39 k
 policycoreutils             x86_64 2.5-33.el7             base           916 k
 policycoreutils-python      x86_64 2.5-33.el7             base           457 k
 python-IPy                  noarch 0.75-6.el7             base            32 k
 scl-utils                   x86_64 20130529-19.el7        base            24 k
 setools-libs                x86_64 3.3.8-4.el7            base           620 k
 xz-devel                    x86_64 5.2.2-1.el7            base            46 k
Updating for dependencies:
 audit-libs                  x86_64 2.8.5-4.el7            base           102 k
 glibc                       x86_64 2.17-292.el7           base           3.6 M
 glibc-common                x86_64 2.17-292.el7           base            11 M
 krb5-libs                   x86_64 1.15.1-37.el7_7.2      updates        805 k
 libcom_err                  x86_64 1.42.9-16.el7          base            41 k
 openssl-libs                x86_64 1:1.0.2k-19.el7        base           1.2 M
 readline                    x86_64 6.2-11.el7             base           193 k

Transaction Summary
================================================================================
Install  19 Packages (+66 Dependent packages)
Upgrade   1 Package  (+ 7 Dependent packages)

Total download size: 126 M
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
http://mirror.as29550.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-binutils-2.27-12.el7.1.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
To address this issue please refer to the below wiki article

https://wiki.centos.org/yum-errors

If above article doesn't help to resolve this issue please use https://bugs.centos.org/.

http://mirror.as29550.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.as29550.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-c%2B%2B-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.as29550.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-libstdc%2B%2B-devel-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.as29550.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-runtime-6.1-1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.sov.uk.goscomb.net/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://repo.uk.bigstepcloud.com/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-binutils-2.27-12.el7.1.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.bytemark.co.uk/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-libstdc%2B%2B-devel-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mozart.ee.ic.ac.uk/CentOS/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-c%2B%2B-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.vorboss.net/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-binutils-2.27-12.el7.1.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://centos.mirrors.nublue.co.uk/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-runtime-6.1-1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.cwcs.co.uk/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirrors.ukfast.co.uk/sites/ftp.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-libstdc%2B%2B-devel-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://centos.mirrors.nublue.co.uk/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-binutils-2.27-12.el7.1.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirrors.ukfast.co.uk/sites/ftp.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-runtime-6.1-1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.mhd.uk.as44574.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.mhd.uk.as44574.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-c%2B%2B-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.sov.uk.goscomb.net/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-binutils-2.27-12.el7.1.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.mhd.uk.as44574.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-libstdc%2B%2B-devel-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://repo.uk.bigstepcloud.com/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-runtime-6.1-1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.bytemark.co.uk/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.sov.uk.goscomb.net/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-c%2B%2B-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mozart.ee.ic.ac.uk/CentOS/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-binutils-2.27-12.el7.1.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.vorboss.net/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-libstdc%2B%2B-devel-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.bytemark.co.uk/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-runtime-6.1-1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mozart.ee.ic.ac.uk/CentOS/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.cwcs.co.uk/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-c%2B%2B-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.mhd.uk.as44574.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-runtime-6.1-1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://repo.uk.bigstepcloud.com/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-c%2B%2B-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://centos.mirrors.nublue.co.uk/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-libstdc%2B%2B-devel-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.cwcs.co.uk/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-binutils-2.27-12.el7.1.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirrors.ukfast.co.uk/sites/ftp.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.sov.uk.goscomb.net/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-runtime-6.1-1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mozart.ee.ic.ac.uk/CentOS/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-libstdc%2B%2B-devel-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.vorboss.net/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-c%2B%2B-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://repo.uk.bigstepcloud.com/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-libstdc%2B%2B-devel-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.vorboss.net/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.bytemark.co.uk/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-c%2B%2B-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.sov.uk.goscomb.net/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-libstdc%2B%2B-devel-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.cwcs.co.uk/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-runtime-6.1-1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://centos.mirrors.nublue.co.uk/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mozart.ee.ic.ac.uk/CentOS/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-runtime-6.1-1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://repo.uk.bigstepcloud.com/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://centos.mirrors.nublue.co.uk/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-c%2B%2B-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.cwcs.co.uk/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-libstdc%2B%2B-devel-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.vorboss.net/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-runtime-6.1-1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirrors.ukfast.co.uk/sites/ftp.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-binutils-2.27-12.el7.1.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.mhd.uk.as44574.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-binutils-2.27-12.el7.1.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirror.bytemark.co.uk/centos/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-binutils-2.27-12.el7.1.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
http://mirrors.ukfast.co.uk/sites/ftp.centos.org/7.7.1908/sclo/x86_64/rh/devtoolset-6/devtoolset-6-gcc-c%2B%2B-6.3.1-3.1.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.


Error downloading packages:
  devtoolset-6-libstdc++-devel-6.3.1-3.1.el7.x86_64: [Errno 256] No more mirrors to try.
  devtoolset-6-runtime-6.1-1.el7.x86_64: [Errno 256] No more mirrors to try.
  devtoolset-6-binutils-2.27-12.el7.1.x86_64: [Errno 256] No more mirrors to try.
  devtoolset-6-gcc-c++-6.3.1-3.1.el7.x86_64: [Errno 256] No more mirrors to try.
  devtoolset-6-gcc-6.3.1-3.1.el7.x86_64: [Errno 256] No more mirrors to try.

/usr/local/src/provision/provision-centos.sh: cannot yum install --assumeyes libxml2-devel libxslt-devel zlib-devel bzip2-devel glibc-devel autoconf bison flex kernel-devel libcurl-devel make cmake openssl-devel libffi-devel libyaml-devel readline-devel libedit-devel bash devtoolset-6-gcc-c++ devtoolset-6-gcc
```
I tracked down that setup of SCL repo causes this and there is no `devtoolset-6` at http://mirror.as29550.net/mirror.centos.org/7.7.1908/sclo/x86_64/rh/ anymore.
Why is gcc-6 rather than 7 necessary?

**Comment by ketan** [comment]
Gcc 6 was needed on centos 6, because the stock gcc was too old for
compiling some native gems/npm packages and the like. This just got
copy/pasted to centos 7.

We can probably use a newer version, if one is available, for centos 6 and
7.

**Comment by marques-work** [comment]
@tomzo @ketan This is fixed as of 2855540381d7dd9be5dfb4031ff9adef5a236dcb.
